src/styles/blocks/_catalog.scss

```diff
@@ -0,0 +1,27 @@
+.catalog {
+  display: grid;
+  grid-template-columns: repeat(4, $card-width);
+  justify-content: center;
+  gap: $card-gap;
+
+  &__card {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: 300px;
+    border: 1px solid #ccc;
+    border-radius: 5px;
+  }
+}
+
+@media (max-width: 1024px) {
+  .catalog {
+    grid-template-columns: repeat(3, $card-width);
+  }
+}
+
+@media (max-width: 768px) {
+  .catalog {
+    grid-template-columns: repeat(2, $card-width);
+  }
+}
```

Comment:
You can also try to set auto-fill to grid-template-columns and set max-width to the main container instead of using media queries